File: javascript/1518.Water-Bottles.js
/**
 * @param {number} numBottles
 * @param {number} numExchange
 * @return {number}
 */
var numWaterBottles = function (numBottles, numExchange) {
  let total = numBottles;

  while (numBottles >= numExchange) {
    const exchange = Math.floor(numBottles / numExchange);
    const remaining = numBottles % numExchange;

    total += exchange;
    numBottles = exchange + remaining;
  }

  return total;
};

File: python/1518.Water-Bottles.py
class Solution:
    def numWaterBottles(self, numBottles: int, numExchange: int) -> int:
        total = numBottles

        while numBottles >= numExchange:
            exchange = numBottles // numExchange
            remaining = numBottles % numExchange

            numBottles = exchange + remaining
            total += exchange

        return total